packages/core/src/error.ts

```diff
@@ -13,14 +15,45 @@ export class ApplicationError extends Error {
   constructor(
     public readonly type: string,
     public readonly status: number,
-    public readonly title: string
+    public readonly title: string,
+    public readonly detail?: string
   ) {
-    super(title)
+    super(detail || title)
   }
 }
 
 export class IllegalStateError extends ApplicationError {
   constructor(description: string) {
-    super("/problem/illegal-state", 500, `Illegal state reached: ${description}`)
+    const { type, code, title } = serverErrorResponse.InternalServerError
+    super(type, code, title, `Illegal state reached: ${description}`)
+  }
+}
+
+export class InternalServerError extends ApplicationError {
```

Comment:
I don't think I agree with these types in the middle. I think each error should probably just pass the ServerErrorRespose.X things manually in the constructor

For example, if ApplicationError takes the entire object

```ts
class EventNotFoundError extends ApplicationError {
  constructor(id: string) {
    super(ServerErrorResponse.InternalServerError, `Event with ID:${id} not found`)
  }
}

Reply:
This of course requires changing the constructor of ApplicationError, but I think in the above suggestion, the code is very readable and understandable